proxy/proxy.go

```diff
@@ -12,27 +12,33 @@ import (
 
 // MakeHTTPClientWithDisableKeepAlives makes a HTTP client with good defaults for timeouts.
 func MakeHTTPClientWithDisableKeepAlives(timeout *time.Duration, tlsInsecure bool, disableKeepAlives *bool) http.Client {
-	if timeout != nil {
-		tr := &http.Transport{
-			Proxy: http.ProxyFromEnvironment,
-			DialContext: (&net.Dialer{
+	client := http.Client{}
+
+	if timeout != nil || tlsInsecure || disableKeepAlives != nil {
+		tr := &http.Transport{Proxy: http.ProxyFromEnvironment}
```

Comment:
Is there a harm in always using this for every client? meaning, could we just use 
```go
client := http.Client{
     Transport: &http.Transport{Proxy: http.ProxyFromEnvironment},
}
```
for all requests?  I walked through the usages of `MakeHTTPClientWithDisableKeepAlives` and `MakeHTTPClient` and all of them set at least one of `timeout` or `disableKeepAlives` to a non-nil value. So it seems that this block will always run and we are always using the `&http.Transport{Proxy: http.ProxyFromEnvironment}` .

Reply:
Personally, reducing complexity is always a good thing